scripts/configure-pipelines.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+
+# Constants
+DEFAULT_RHDH_DEPLOYMENT="backstage-ai-rh-developer-hub" # deployment created by rhdh operator by default
+PLUGIN_CONFIGMAP="backstage-dynamic-plugins-ai-rh-developer-hub" # configmap created by rhdh operator for plugins by default
+CRD="tektonconfigs"
+
+# Variables
+BASE_DIR="$(realpath $(dirname ${BASH_SOURCE[0]}))/.."
+RHDH_DEPLOYMENT="${DEFAULT_RHDH_DEPLOYMENT}"
+RHDH_PLUGINS_CONFIGMAP="${PLUGIN_CONFIGMAP}"
+NAMESPACE=${NAMESPACE:-"ai-rhdh"}
+PIPELINES_NAMESPACE=${PIPELINES_NAMESPACE:-"openshift-pipelines"}
+PIPELINES_SECRET_NAME=${PIPELINES_SECRET_NAME:-'rhdh-pipelines-secret'}
+EXISTING_NAMESPACE=${EXISTING_NAMESPACE:-''}
+EXISTING_DEPLOYMENT=${EXISTING_DEPLOYMENT:-''}
+RHDH_PLUGINS=${RHDH_PLUGINS:-''}
+RHDH_INSTANCE_PROVIDED=${RHDH_INSTANCE_PROVIDED:-false}
+
+# Secret variables
+GITHUB__APP__ID=${GITHUB__APP__ID:-''}
+GITHUB__APP__WEBHOOK__SECRET=${GITHUB__APP__WEBHOOK__SECRET:-''}
+GITHUB__APP__PRIVATE_KEY=${GITHUB__APP__PRIVATE_KEY:-''}
+GITOPS__GIT_TOKEN=${GITOPS__GIT_TOKEN:-''}
+GITLAB__TOKEN=${GITLAB__TOKEN:-''}
+QUAY__DOCKERCONFIGJSON=${QUAY__DOCKERCONFIGJSON:-''}
+
+# Use existing variables if RHDH instance is provided
+if [[ $RHDH_INSTANCE_PROVIDED != "true" ]] && [[ $RHDH_INSTANCE_PROVIDED != "false" ]]; then
+    echo -n "RHDH_INSTANCE_PROVIDED needs to be set to either 'true' or 'false'"
+    echo "FAIL"
+    exit 1
+elif [[ $RHDH_INSTANCE_PROVIDED == "true" ]]; then
+    NAMESPACE="${EXISTING_NAMESPACE}"
+    RHDH_DEPLOYMENT="${EXISTING_DEPLOYMENT}"
+    RHDH_PLUGINS_CONFIGMAP="${RHDH_PLUGINS}"
+fi
+
+# Reading secrets
+# Reads secrets either from environment variables or user input
+echo "* Reading secrets: "
+# Reads GitHub Org App ID
+until [ ! -z "${GITHUB__APP__ID}" ]; do
+    read -p "Enter your GitHub App ID: " GITHUB__APP__ID
+    if [ -z "${GITHUB__APP__ID}" ]; then
+        echo "No GitHub App ID entered, try again."
+    fi
+done
+
+# Reads GitHub Org App Webhook Secret
+# Optional: If left blank during user prompt, one is generated instead
+if [ -z "${GITHUB__APP__WEBHOOK__SECRET}" ]; then
+    read -p "Enter your GitHub App Webhook Secret (Optional): " GITHUB__APP__WEBHOOK__SECRET
+    if [ -z "${GITHUB__APP__WEBHOOK__SECRET}" ]; then
+        GITHUB__APP__WEBHOOK__SECRET="$(openssl rand -hex 20)"
+        echo "Use the following as your GitHub App Webhook Secret: ${GITHUB__APP__WEBHOOK__SECRET}"
+    fi
+fi
+
+# Reads GitHub Org App Private Key
+until [ ! -z "${GITHUB__APP__PRIVATE_KEY}" ]; do
+    read -p "Enter your GitHub App Private Key (Use CTRL-D when finished): " -d $'\04' GITHUB__APP__PRIVATE_KEY
+    if [ -z "${GITHUB__APP__PRIVATE_KEY}" ]; then
+        echo "No GitHub App Private Key entered, try again."
+    fi
+done
+
+# Reads Git PAT
+# Optional: If left blank during user prompt, the namespace secret will not be created
+if [ -z "${GITOPS__GIT_TOKEN}" ]; then
```

Comment:
Nit: When doing the script and manually inputting the values via CLI the end of the private key and the gitops prompt are on the same line:

```
-----END RSA PRIVATE KEY-----^DEnter your Git Token (Optional): <redacted token>
```

Reply:
Done in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/7/commits/6d03b6bd7ba3a8f9aaf80253a88ed23d2518dee8